### ISSUE TITLE
2021 Roadmap

### ISSUE THREAD

**Issue by Chalarangelo** [30-seconds/30-seconds-web#208]
2021 is a little bit more than 1 month away, so we should start planning. Contrary to 2020, that we followed the OKR and Agile methodology, we are taking a different approach in 2021. Here's the rundown:

### Goal-setting and planning

Setting goals at the beginning of a quarter has been successful for the most part. However, we found that we miss out on potential opportunities that might arise mid-quarter when using this approach. Instead of a quarterly dissection of 2021, we will plan on a monthly basis, adjusting goals to match current events. This should help us be significantly more focused and deliver more consistent updates.

### Goal categorization and actionable fields

Planning during 2020 was roughly split into **Branding**, **UI/UX**, **Technology** and **Content**. Moving forward we will keep this kind of goal segmentation, however we will define each one more clearly to allow us to better track progress in each field. The plan is to break down the year into roughly 25% each, so that we can focus on all areas. As it currently stands, some areas (i.e. **UI/UX** and **Content**) are more valuable than others so the breakdown might not be 100% even between all 4. Goals, milestones and even tasks should be tagged appropriately to help us better track progress in each field.

### Sprint duration

Sprints have been mostly a success, from the way we separate tasks to motivating us to actually complete them. Instead of the 3-week schedule, we'll match sprints with monthly plans, so each Sprint will be 4 weeks long (1 month). This will help us set up a better flow and match goals.

### Task organization

The general-purpose 2020 board was pretty useful, but became chaotic over time. We'll maintain an internal board next year, so that all web-related tasks are in this repo. Milestones have been very useful in planning, so we'll create monthly/sprint milestones in 2021 and track goals same as this year. In addition to that, the quarterly issues and milestones will be leaving us and we will use this thread to track the annual progress instead.

### Task priority and duration

We want to be able to plan tasks more carefully, so we are going to split everything into really small chunks of work, delivering smaller updates more often. The most likely scenario is that we will see a lot of new issues added to the project and tackled with more frequent releases, while also using labels to track priority and task size. More on that later.

### Release cycle

Instead of many releases, we'll opt for monthly releases to keep track of major changes. This will affect the feature branch flow and also the release cycle. Major releases should be roughly twice a year and minor releases should be roughly once a month. Patches might release more frequently and with less specific timeframes.

---

More information coming soon. Cheers!

**Comment by Chalarangelo** [comment]
### Proposed goals

In order to be able to plan a little ahead and know what might be of interest to us, we'll try to maintain a pool of goals that we can implement in 2021. This is by no means a full list and some of these goals might not be actively worked on during the year:

- **Create a homepage that acts as a starting point for user journeys.** User journeys are more varied than the ones we launched with back in January 2019. We should create a homepage that promotes discoverability while presenting our digital product more concisely to new users. This might include some featured categories, a link to a categories page, some featured snippets and an intro with some illustrations, as well as a more prominent search area.
- **Create curated content experiences.** Collections of snippets can help promote discoverability, while also aiding users in their journey and helping them figure out more complex tasks. We should start accommodating this as soon as we can.
- **Integrate conventional commits and changelogs in our workflow.** By providing changelogs and using conventional commits with a tool like commitizen to do so, we could better track changes to the website and inform users about new milestones.

**Comment by Chalarangelo** [comment]
### January

**Milestone:** [2021/01](https://github.com/30-seconds/30-seconds-web/milestone/21)
**Categories:** UI/UX
**Focus:** Content discovery and display
**Description:** We are working towards better content discoverability, improved user experience and general improvements in content display. To that end, we are making changes in literals, content display, styles, designs etc. as well as making an effort to introduce a better "hub" homepage, a collection system for curated user journeys and other improvements to go with them.
**Notes:** We are not 100% confident these changes will be fully implemented in this sprint, however we expect all of the to be in full effect by the end of next sprint (meaning some tasks might be left over for the next sprint).

**Comment by Chalarangelo** [comment]
#### Alexa data (Jan. 4th, 2021)

![image](https://user-images.githubusercontent.com/8281875/103510834-a0701080-4e6e-11eb-80b7-413d08947a1d.png)

**Comment by Chalarangelo** [comment]
#### Alexa data (Jan. 19th, 2021)

![image](https://user-images.githubusercontent.com/8281875/104998671-4a859600-5a34-11eb-9246-ccd09a89d774.png)

**Comment by Chalarangelo** [comment]
### Hotjar Heatmap Results

We run a **Heatmap** via **Hotjar** on the homepage after launching **stage 2 of the collections**. Here are the results compared to a previous Heatmap from September:

|Interaction|Clicks/User (current)|Clicks/user (previous)|Delta|
|---|---|---|---|
|JS|0.32|0.16|+100%|
|CSS|0.11|0.09|+22%|
|React|0.07|0.06|+16%|
|Python|0.05|0.05|+0%|
|Articles|0.03|0.03|+0%|
|Snippets|0.03|0.1|-70%|
|Search|0.03|0.04|-25%|
|View all categories|0.08|N/A|N/A|

#### Notes

- The decrease in interactions with the main snippet list is both desired and expected, as we have demoted them in the home page and also made them a smaller section. The whole point of collections was providing more curated experiences tailored to users' needs, which means our focus is on the top categories more than the main listing.
- The decrease in search might also be accounted for by the categories redesign, even though the absolute difference is around the margin of statistical error and could be ignored.
- The first few categories that appear on the page are far more interacted with, netting us upwards to twice the same traffic towards them, whereas categories later in the list are around the same as before.

#### Conclusion

The result is very close to our original expectations, showing that users opt for narrower, more focused experiences based on their needs. Therefore, collections seem like a successful feature launch that we should keep on expanding to better cater to our users.

**Comment by Chalarangelo** [comment]
### January Retro

[TODO]

**Comment by Chalarangelo** [comment]
### February

**Milestone:** [2021/02](https://github.com/30-seconds/30-seconds-web/milestone/22)
**Categories:** UI/UX, Technology, Content
**Focus:** Content discovery, content assortment, content management
**Description:** As January's goals have been delivered mostly in time, we are taking February to follow up on any leftover maintenance tasks. We are also going to tackle the autocomplete feature that was promised for January and add more content, mainly in the form of articles. While working on those features, we are going to take a look at the content management system and handle some of the emerging issues in managing content at large.
**Notes:** As this sprint contains a lot of separate smaller goals, some might not be delivered in time or even be postponed for the next sprint. The main focus UI/UX-wise should be the autocomplete feature, which should take up about 50% of the sprint, allowing us to work on other things as well. Tech needs might take a back seat, while we are adding more content, scheduling posts for March and April if possible to allow us to move forward with some larger tasks over the next couple of sprints.

**Comment by Chalarangelo** [comment]
#### Google Search Console Performance (Jan. 27th, 2021)
![image](https://user-images.githubusercontent.com/8281875/106139330-f6865a00-6175-11eb-8280-09f1071aa902.png)

Search clicks on Jan. 27th, 2021 reached an all-time high of **1170 clicks/day**.

**Comment by Chalarangelo** [comment]
Alexa data (Feb. 1st, 2021)

![image](https://user-images.githubusercontent.com/8281875/106448212-20e45a00-648b-11eb-9de3-5a87d2ab0c0f.png)

**Comment by Chalarangelo** [comment]
#### Analytics data (Jan 24 - 30, 2021 compared to Jan 17 - 23, 2021)

![image](https://user-images.githubusercontent.com/8281875/106518435-8f083b80-64e2-11eb-96fe-1576224497e9.png)

**Comment by Chalarangelo** [comment]
On an organizational note, we just added labels for task size with numeric (1-5) and descriptive values. These should be used to approximate the amount of tasks that can be completed in any given sprint.

**Comment by Chalarangelo** [comment]
![image](https://user-images.githubusercontent.com/8281875/107237638-9e7e0c00-6a2f-11eb-9409-74a341b0b330.png)

**Comment by Chalarangelo** [comment]
#### Alexa data (Feb. 10th, 2021)

![image](https://user-images.githubusercontent.com/8281875/107511115-b54a6d00-6bad-11eb-817a-32a148b247d0.png)

**Comment by Chalarangelo** [comment]
#### Ubersuggest report (Feb. 10th, 2021)

![image](https://user-images.githubusercontent.com/8281875/107512136-132b8480-6baf-11eb-94a2-31f3fe0f78d2.png)

**Comment by Chalarangelo** [comment]
#### Alexa data (Feb. 22nd, 2021)

![image](https://user-images.githubusercontent.com/8281875/108718674-5da1ef00-7527-11eb-9ec5-ec7568a57a7c.png)

**Comment by Chalarangelo** [comment]
### February Retro

[TODO]

**Comment by Chalarangelo** [comment]
### march

**Milestone:** [2021/03](https://github.com/30-seconds/30-seconds-web/milestone/23)
**Categories:** UI/UX, Content
**Focus:** Dark mode, new content
**Description:** February was mostly successful in terms of content discoverability and dealing with certain tech needs. March will be focused on user experience and revamping the dark mode as the primary presentation experience for the website from here onwards. We should also focus on creating new content to schedule for the future.
**Notes:** This sprint should focus on some major changes in the user experience and content. While these are just a handful of tasks by themselves, we might have some carryover from the previous sprint, so we can cover some of the spare time with leftover tasks.

**Comment by Chalarangelo** [comment]
#### Google Search Console Performance (Feb. 27th, 2021)

![image](https://user-images.githubusercontent.com/8281875/109383222-cdc8c000-78ed-11eb-9030-4f6503c25ce6.png)

Search clicks on Feb. 24th, 2021 reached an all-time high of 1237 clicks/day.

**Comment by Chalarangelo** [comment]
Google Search Console Performance (Mar. 14th, 2021)

![image](https://user-images.githubusercontent.com/8281875/111070695-7de81c80-84db-11eb-81db-58cec2343936.png)

Search clicks on Mar. 9th, 2021 reached an all-time high of **1429 clicks/day**, a 13% increase since the last report.

**Comment by Chalarangelo** [comment]
#### Alexa data (Mar. 14th, 2021)


![image](https://user-images.githubusercontent.com/8281875/111070761-cf90a700-84db-11eb-9cb8-0dde99df7e60.png)

**Comment by Chalarangelo** [comment]
On an infrastructure note, we migrated from Travis CI to GitHub Actions for continuous integration. This is mainly due to limitations and costs with Travis CI. GitHub Actions was the optimal alternative as it integrates well with our current workflow and required minimal setup.

**Comment by Chalarangelo** [comment]
#### Google Search Console Performance (Mar. 21st, 2021)

![image](https://user-images.githubusercontent.com/8281875/111898158-fba4ae80-8a2c-11eb-83c0-fbdbb6cdc1b9.png)

Search clicks on Mar. 16th, 2021 reached an all-time high of **1757 clicks/day**, a 22% increase since the last report.

Interestingly enough, average position has improved over time (16.7 in late January compared to 16.1 in late March).

**Comment by Chalarangelo** [comment]
#### Core Web Vitals Report (Mar. 24th, 2021)

![image](https://user-images.githubusercontent.com/8281875/112276276-3acf3b80-8c89-11eb-9ea0-c7b09a07aa76.png)

**Comment by Chalarangelo** [comment]
### March Retro

March run into some issues due to IRL pressures. We managed to deliver the dark mode revamp and enough content to cover the whole month, so nothing went off track there. 

Generally speaking, even with some external factors rooting against us, we still delivered enhancements and content in time without major hiccups. If there's a takeaway from this is that the current sprint structure suits us well and is able to handle situations like this well.

The actionables we should set for future sprints are focused on content creation and planning way ahead in terms of content to ensure no IRL factors can throw us off schedule. Apart from that, it seems like most of the platform-related changes are coming to a close, so content should be our focus for a while and any changes should be made to suit the content creation workflow as soon as possible.

**Comment by Chalarangelo** [comment]
### April

**Milestone:** [2021/034](https://github.com/30-seconds/30-seconds-web/milestone/24)
**Categories:** Infrastructure
**Focus:** Backend maintenance
**Description:** Our task backlog has been growing for a while and we have a lot of leftover maintenance tasks that need handling and evaluation. We're going to focus on those, researching potential solutions and pursuing them as necessary. Deliverables are going to be more varied but mostly infrastructure-related in order to unblock future changes.
**Notes:** As this sprint is mostly a maintenance one, we can focus on content in parallel with tasks. Content creation is always a focus, so there's no reason to repeat ourselves. Thankfully, there's no carry-over from the previous sprint - if anything, it's the opposite - so we can more easily tidy up our backlog.

**Comment by Chalarangelo** [comment]
#### Alexa data (Apr. 5th, 2021)

![image](https://user-images.githubusercontent.com/8281875/113563661-c2fdfb00-9610-11eb-8d62-23e8fd2019ce.png)

**Comment by Chalarangelo** [comment]
30 seconds of code was featured on [Smashing Magazine](https://www.smashingmagazine.com/2021/04/vanilla-javascript-code-snippets/) on April 5th, 2021.
![image](https://user-images.githubusercontent.com/8281875/113574908-ba63ef80-9625-11eb-8135-142881396b6f.png)

**Comment by Chalarangelo** [comment]
We were the 2nd top trending project in JavaScript on GitHub for the past week:
![image](https://user-images.githubusercontent.com/8281875/113581316-a9b87700-962f-11eb-8bfc-e518313ac74a.png)

**Comment by Chalarangelo** [comment]
We are currently number 1 result for the query "JavaScript Snippets" on Google Search! 🎉 

![image](https://user-images.githubusercontent.com/8281875/113674246-aa4e1d80-96c2-11eb-9d87-0ad06e439df7.png)

**Comment by Chalarangelo** [comment]
#### Ubersuggest report (Apr. 7th, 2021)

![image](https://user-images.githubusercontent.com/8281875/113825107-3fb3e500-9789-11eb-895c-b0074c226f7a.png)

**Comment by Chalarangelo** [comment]
30 seconds of code recently featured in [JavaScript Weekly Issue 533](https://javascriptweekly.com/issues/533) via the latest Smashing Magazine article we posted above.

![image](https://user-images.githubusercontent.com/8281875/114278757-ea522f00-9a39-11eb-964b-277a1f6afb6b.png)

**Comment by Chalarangelo** [comment]
#### Google Search Console Performance (Apr. 13th, 2021)

![image](https://user-images.githubusercontent.com/8281875/114504924-48e4fc00-9c38-11eb-9aae-24a1b1fe29a9.png)

Search clicks on Apr. 7th, 2021 reached an all-time high of 2081 clicks/day, a 18% increase since the last report.

**Comment by Chalarangelo** [comment]
#### Content repository release

Today, Apr. 17th, 2021, 30 seconds of Git is released with 80+ original snippets.

**Comment by Chalarangelo** [comment]
30 seconds of code is the 5th most trending repository today with over 300 stars.

![image](https://user-images.githubusercontent.com/8281875/115302658-584dd300-a16b-11eb-8089-780382082c80.png)

**Comment by Trinityyi** [comment]
![image](https://user-images.githubusercontent.com/13429517/115305819-951bc900-a16f-11eb-89d4-a487e23fcec1.png)
Shared on facebook by Constwardtz

**Comment by Chalarangelo** [comment]
#### Google Search Console Page Experience (Apr. 20th, 2021)


![image](https://user-images.githubusercontent.com/8281875/115384680-fc716180-a1df-11eb-90d1-b57e47d8de08.png)

**Comment by Chalarangelo** [comment]
### April Retro

April was a very productive month. We delivered a lot of infrastructure and tooling updates, while we also managed to start researching and actively developing towards a Next.js-based solution for the platform.

April also saw the release of the Git content repository with over 80 snippets, whereas the flow of new articles has been constant. SEO and community have been extremely kind to us, helping us reach higher peaks in traffic and engagement. 

In summary, April was just about as good as it gets in terms of progress towards a better product. As far as actionables go, we only need to figure out what exactly we did right this sprint and refine upon it to make it even better. The recipe seems to be a lot of smaller technical changes and feature releases, while creating content at a stable pace.

**Comment by Chalarangelo** [comment]
### May

**Milestone:** [2021/05](https://github.com/30-seconds/30-seconds-web/milestone/25)
**Categories:** TBD
**Focus:** TBD
**Description:** TBD

**Comment by Chalarangelo** [comment]
Technical note: The migration from Gatsby to Next.js is finally complete. From what we can tell, there were no hiccups in the process and everything is working as expected. We'll continue to monitor and patch any issues in the following days.

**Comment by Chalarangelo** [comment]
![image](https://user-images.githubusercontent.com/8281875/116982394-648c7100-acd1-11eb-918b-bf2d787719f8.png)

April was our best month in the last year in terms of GitHub star increase.

**Comment by Chalarangelo** [comment]
#### Alexa data (May. 4th, 2021)

![image](https://user-images.githubusercontent.com/8281875/116982519-8ede2e80-acd1-11eb-91c9-e570d89716ca.png)

**Comment by Chalarangelo** [comment]
#### Alexa data (May. 11th, 2021)

![image](https://user-images.githubusercontent.com/8281875/117768015-6c509600-b23a-11eb-82af-34a516900a0d.png)

**Comment by Chalarangelo** [comment]
![image](https://user-images.githubusercontent.com/8281875/117768213-bd608a00-b23a-11eb-801b-17212ac7f14c.png)

30 seconds of code is in the top 3 trending JS repos today.

**Comment by Chalarangelo** [comment]
#### Alexa data (May 21st, 2021)
![image](https://user-images.githubusercontent.com/8281875/119187544-2b4e5200-ba82-11eb-96c4-2cb405cf4644.png)

**Comment by Chalarangelo** [comment]
Google Search Console Performance (May 23rd, 2021)

![image](https://user-images.githubusercontent.com/8281875/119256888-8049af80-bbcb-11eb-98b9-591cb914744e.png)

No substantial change in search result performance, we had a bit of a tumble in past weeks, but we seem to be climbing back up right now.

**Comment by Chalarangelo** [comment]
![image](https://user-images.githubusercontent.com/8281875/119385221-84122a80-bcce-11eb-95d8-d1dd5b33f23a.png)

More than 40% of the website's traffic the past week came from Google Search.

**Comment by Chalarangelo** [comment]
### May Retro

May went fairly well, with Next.js being released without any hiccups and small improvements arriving over the month. Content was delivered on time and everything was generally quite smooth.

Actionables are again going to be focused on content creation and planning ahead, so we're just reiterating past sprints at this point.

**Comment by Chalarangelo** [comment]
### June

**Milestone:** [2021/06](https://github.com/30-seconds/30-seconds-web/milestone/26)
**Categories:** Branding
**Focus:** Rebranding for the new 30 seconds era
**Description:** This sprint will be mainly focused on rebranding the project with an eye to the future. Design and UI changes will be the main focus, as well as delivering a consistent brand identity by the end of the month.
**Notes:** Content will be the less visible priority during this sprint. We aim to backdate a lot of articles for the summer months and it's a good time to start doing so.

**Comment by Chalarangelo** [comment]
![image](https://user-images.githubusercontent.com/8281875/120080261-afc35500-c0c0-11eb-8d37-f37d61bdd7a9.png)

**Comment by Chalarangelo** [comment]
![image](https://user-images.githubusercontent.com/8281875/120188175-786aba80-c21e-11eb-924d-87606315abb2.png)

**Comment by Chalarangelo** [comment]
#### Alexa data (Jun. 8th, 2021)
![image](https://user-images.githubusercontent.com/8281875/121129301-6c41b700-c835-11eb-8bc6-f4c5ba45f80c.png)

**Comment by Chalarangelo** [comment]
#### Ubersuggest data (Jun. 8th, 2021)

![image](https://user-images.githubusercontent.com/8281875/121129503-bb87e780-c835-11eb-9a22-2acb99425f9a.png)
![image](https://user-images.githubusercontent.com/8281875/121129534-c6427c80-c835-11eb-807f-3503252061a8.png)

**Comment by Chalarangelo** [comment]
#### Search Console Insights (Jun. 17, 2021)

![image](https://user-images.githubusercontent.com/8281875/122341896-a8b79600-cf4c-11eb-89f6-23c1246912b3.png)
![image](https://user-images.githubusercontent.com/8281875/122341939-b3722b00-cf4c-11eb-8aae-b9c2071fe970.png)
![image](https://user-images.githubusercontent.com/8281875/122341964-bcfb9300-cf4c-11eb-9ffd-939c1cb11af8.png)
![image](https://user-images.githubusercontent.com/8281875/122342051-d270bd00-cf4c-11eb-9510-f57a17c7ac67.png)
![image](https://user-images.githubusercontent.com/8281875/122342084-da306180-cf4c-11eb-92cd-6b84e69099d5.png)
![image](https://user-images.githubusercontent.com/8281875/122342126-e61c2380-cf4c-11eb-9be2-175cb4e9a901.png)
![image](https://user-images.githubusercontent.com/8281875/122342161-f207e580-cf4c-11eb-96c1-9b0218298e91.png)

**Comment by Chalarangelo** [comment]
We just launched the rebranded website alongside the new logo, a host of new features and a lot of other changes. Social media updated, too! :tada:

**Comment by Chalarangelo** [comment]
![image](https://user-images.githubusercontent.com/8281875/122954380-0eaa8000-d388-11eb-85e6-1fc5da8f7771.png)

30 seconds of code is trending on GitHub.

**Comment by Chalarangelo** [comment]
#### Alexa data (Jun. 28th, 2021)

![image](https://user-images.githubusercontent.com/8281875/123588986-3fa80c00-d7f1-11eb-8165-ae39038b0735.png)

**Comment by Chalarangelo** [comment]
### June Retro

June was a great month both in terms of content and infrastructure work and it also marks the start of the summer season. Most of the backlog was cleared, major and minor infrastructure and maintenance tasks were completed, while we also managed to deliver the rebranding on time and without any hiccups. Content delivery was already streamlined and it got even better with the new article infrastructure updates, resulting in an overall more efficient article creation pipeline. 

Overall, pacing was great and all actionables were finished earlier than expected and without issues. In terms of actionables, there are very few things I would change. However, due to the context switch that will occur during the summer months (i.e. focusing on content creation), we should be on our toes for any issues that arise due to that in the coming sprints.

**Comment by Chalarangelo** [comment]
### July

**Milestone:** [2021/07](https://github.com/30-seconds/30-seconds-web/milestone/27)
**Categories:** Content
**Focus:** Creating enough content for the summer months
**Description:** Summer is here, which means we're taking a much needed break from infrastructure changes. We'll keep creating content and scheduling it to go live over the summer, so that's our sole focus in July. We should create enough content to cover all of July and August with articles (3 per 2 weeks instead of the usual 4) and add any snippets we might be able to fit into our schedule.
**Notes:** Infrastructure and maintenance may occur, but not as often and will not be a priority for this sprint.

**Comment by Chalarangelo** [comment]
#### Google Search Console Page Experience (Jul. 2nd, 2021)

![image](https://user-images.githubusercontent.com/8281875/124226211-67f57a80-db11-11eb-9dee-dde4dff91456.png)

**Comment by Chalarangelo** [comment]
#### Alexa data (Jul. 10th, 2021)

![image](https://user-images.githubusercontent.com/8281875/125155799-dd44f900-e16a-11eb-9f62-415a2f9a9101.png)

**Comment by Chalarangelo** [comment]
![image](https://user-images.githubusercontent.com/8281875/125173096-f0d17d80-e1c5-11eb-90bf-2b0e9024096f.png)

 30 seconds of code is trending on GitHub.

**Comment by Chalarangelo** [comment]
![image](https://user-images.githubusercontent.com/8281875/125940846-788c677b-899d-4305-a019-48f909807388.png)

30-seconds-of-code is the 2nd fastest growing repo on **Best of JS** in the past 12 months.

**Comment by Chalarangelo** [comment]
### July Retro

July was one of our slowest months in terms of product updates, due to a much needed summer break. Regardless, we delivered crucial styling and maintenance updates that will help polish the user experience without putting too much effort into these changes. Content creation, on the other hand, was increased significantly, allowing us to schedule content for the coming month and enjoy our summer break.

Considering this is the first summer break since we introduced most of these systems and the latest planning framework, everything went exactly as we'd hoped. No actionables or takeaways from this sprint, hoping the next one follows suit.

**Comment by Chalarangelo** [comment]
### August

**Milestone:** [2021/08](https://github.com/30-seconds/30-seconds-web/milestone/28)
**Categories:** Content
**Focus:** Improving upon collections, planning for the future
**Description:** Second sprint of the summer break period, so little to no time will be left for infrastructure changes. Minor changes may happen, some tinkering and experimenting, but no hard deadlines. The spotlight will be once again on the content front, mainly planning towards future content and collections as well as creating collections from existing content. A minor enhancement could potentially land on the collections page, allowing us to paginate it to create a smoother user experience.
**Notes:** Very little work to be done during this sprint, even if there's zero effort put towards the project, there shouldn't be any issues whatsoever.

**Comment by Chalarangelo** [comment]
#### Google Search Console Page Experience (Jul. 22nd, 2021)


![image](https://user-images.githubusercontent.com/8281875/126616047-646ea701-a911-405f-8d13-62b0f77a8a81.png)

**Comment by Chalarangelo** [comment]
![image](https://user-images.githubusercontent.com/8281875/126617302-083623e8-ec8a-40a0-9a40-6bb64c576666.png)

30 seconds of code was featured on [1stWebDesigner](https://1stwebdesigner.com/6-resources-for-javascript-code-snippets/)

**Comment by Chalarangelo** [comment]
![image](https://user-images.githubusercontent.com/8281875/129470742-3a642498-9940-4b1a-90ed-e2ae96be1de9.png)

The past few articles on JS Data Structures seem to perform very well on Daily Dev.

**Comment by Chalarangelo** [comment]
#### Google Search Console Data (Aug. 15th, 2021)

![image](https://user-images.githubusercontent.com/8281875/129470806-754b1fc2-5fdf-4cb6-afe1-eaa042562283.png)

![image](https://user-images.githubusercontent.com/8281875/129470968-6971ec11-ef7b-4723-98dc-cd7db082785f.png)

![image](https://user-images.githubusercontent.com/8281875/129470993-a8345b0a-ae31-4c7c-8d57-9a455c699662.png)

![image](https://user-images.githubusercontent.com/8281875/129470996-a64eb30c-c7cf-4ca8-8c1c-ccf9fb480685.png)

![image](https://user-images.githubusercontent.com/8281875/129471031-946bc69d-642f-4ccb-b146-44563e165b9c.png)

![image](https://user-images.githubusercontent.com/8281875/129471054-7da016a4-963d-484f-9c08-8bcae027ec05.png)




- Search clicks on **Aug. 9th, 2021** reached an all-time high of 2167 clicks/day, a 4% increase since the last report.
- 30 seconds of code has started appearing on **Discover**.
- New content is reasonably varied in terms of performance, but we still see some pieces getting significant traffic. Existing content popularity remains unchanged.
- Social and search traffic are pretty much same as a couple of months back, search performance has dipped slightly.
- Twitter seems to be our most valued social media (makes sense since it's the only one we actually use).
- Page experience seems very fiddly in terms of data.

**Comment by Chalarangelo** [comment]
#### Alexa data (Aug. 15th, 2021)

![image](https://user-images.githubusercontent.com/8281875/129471079-adad7666-6d75-4c62-a7e3-705b7ac015cc.png)

**Comment by Chalarangelo** [comment]
#### Ubersuggest data (Aug. 15th, 2021)

![image](https://user-images.githubusercontent.com/8281875/129471146-5ec60abe-cbb0-4213-84a3-40d32273fdc8.png)
![image](https://user-images.githubusercontent.com/8281875/129471187-d3ef14b0-a7bf-4ffe-9798-d1f134f87f19.png)
![image](https://user-images.githubusercontent.com/8281875/129471200-3bb89a3c-1905-49ca-912d-33e2f9f3ba67.png)
![image](https://user-images.githubusercontent.com/8281875/129471233-a81dad29-12d3-4354-bd21-d450aab632ea.png)
![image](https://user-images.githubusercontent.com/8281875/129471239-78afdb77-7d1a-4983-bfc0-7b5a28fa4622.png)

**Comment by Chalarangelo** [comment]
![image](https://user-images.githubusercontent.com/8281875/129471298-d04d944e-cfc2-46e8-b26a-10ca841f805d.png)

**Comment by Chalarangelo** [comment]
#### Alexa data (Aug. 18th, 2021)


![image](https://user-images.githubusercontent.com/8281875/129843254-65868a8c-a614-4409-bfaa-bb442b28c063.png)

**Comment by Chalarangelo** [comment]
#### Alexa data (Aug. 22nd, 2021)

![image](https://user-images.githubusercontent.com/8281875/130351585-97060d7e-37d1-406b-8093-92f3ebda5a2c.png)

**Comment by Chalarangelo** [comment]
#### Alexa data (Aug. 30th, 2021)

![image](https://user-images.githubusercontent.com/8281875/131290685-1bc09e1c-f78f-4189-9dcc-2de0a77d5496.png)

**Comment by Chalarangelo** [comment]
#### August Retro

August was a summer break month with little to no platform and product updates. However, content rollout was steady and planned well and gave us enough space to get back into content creation before scheduled posts were finished publishing. 

While little work was done this month, everything leading up to this summer break was planned and executed perfectly, leaving us some much needed vacation time. Let's repeat this next year!

**Comment by Chalarangelo** [comment]
### August

**Milestone:** [2021/09](https://github.com/30-seconds/30-seconds-web/milestone/29)
**Categories:** Content, Technology
**Focus:** Additional collections, articles, snippets, infrastructure updates
**Description:** Back from the summer break period, our main focus right off the bat is to schedule content and create new collections as we start developing our content strategy and looking into the future. While content will be our core focus yet again, some platform and infrastructure updates should take place in parallel to make sure everything is up to date, fast and secure.
**Notes:** Next.js update should be one of our focus goals for this sprint, whereas we should start looking into outdated dependencies, too. On the content front, we have to keep investing in content strategy and competitive research to better figure out what our readers want to read.

**Comment by Chalarangelo** [comment]
### Ubersuggest data (Sep. 2nd, 2021)

![image](https://user-images.githubusercontent.com/8281875/131965471-2bb323b0-3ee6-4931-be3d-9a339931ba3f.png)
![image](https://user-images.githubusercontent.com/8281875/131965595-f659729f-5964-4359-9cc4-95553c679e35.png)
![image](https://user-images.githubusercontent.com/8281875/131965728-e63c2d71-9d2f-424c-84ca-d47cb83ccb74.png)

**Comment by Chalarangelo** [comment]
#### Google Search Console Discover Data (Sep. 2nd, 2021)

![image](https://user-images.githubusercontent.com/8281875/131966118-281a9ac3-5836-4660-955a-f8e9674366ef.png)

**Comment by Chalarangelo** [comment]
30 seconds of code was recently featured in [Developer Updates](https://www.developerupdates.com/blog/5-useful-websites-every-website-developer-should-know)

![image](https://user-images.githubusercontent.com/8281875/132175952-8f94f1f8-0927-473d-8b20-63c12b75735b.png)

**Comment by Chalarangelo** [comment]
[Popular tweet about 30 seconds of code](https://twitter.com/VladPasca5/status/1434493072801517574?s=20)

![image](https://user-images.githubusercontent.com/8281875/132205469-ce32bda1-a7f3-4eae-ab87-d146886a6d8f.png)

**Comment by Chalarangelo** [comment]
![image](https://user-images.githubusercontent.com/8281875/132222672-61e1d551-5a1e-4ac7-8382-37407dd1cf0d.png)

30 seconds of code is trending on GitHub.

**Comment by Chalarangelo** [comment]
### August Search Performance

![image](https://user-images.githubusercontent.com/8281875/132293154-e1628ed3-18f7-49c6-be59-69c39c078775.png)
![image](https://user-images.githubusercontent.com/8281875/132293189-22d71059-3dfd-40d7-ab94-de60ec4b07b5.png)
![image](https://user-images.githubusercontent.com/8281875/132293224-423843f5-0898-4efa-b912-4fcb8cf00720.png)
![image](https://user-images.githubusercontent.com/8281875/132293271-b2779947-0b89-42cf-8181-47074752367e.png)

**Comment by Chalarangelo** [comment]
![image](https://user-images.githubusercontent.com/8281875/132295329-3e495913-3e76-44ae-89e3-c774f842b797.png)

30 seconds of code was featured on [Designership](https://thedesignership.com/resource/30-seconds-of-code/)

**Comment by Chalarangelo** [comment]
![image](https://user-images.githubusercontent.com/8281875/132664663-fce8ba58-154e-4639-95b5-d055b7d33c46.png)

30 seconds of code is now the featured snippet for the search term `JavaScript algorithms`.

**Comment by Chalarangelo** [comment]
#### Google Search Console Performance (Sep. 9th, 2021)

![image](https://user-images.githubusercontent.com/8281875/132721401-c702385a-5c83-4e50-b1b6-da7c139d3101.png)

Search clicks on Sep. 7th, 2021 reached an all-time high of 2359 clicks/day, a 10% increase since the last report.

**Comment by Chalarangelo** [comment]
![image](https://user-images.githubusercontent.com/8281875/133905131-298b296b-a775-480d-a009-43d62ebd8ea5.png)

30 seconds of code is the top trending JavaScript repository today!

**Comment by Chalarangelo** [comment]
#### Alexa data (Sep. 21st, 2021)

![image](https://user-images.githubusercontent.com/8281875/134119438-2d9d2d10-16b6-401f-90e2-8a1ca97c7dc8.png)

**Comment by Chalarangelo** [comment]
![image](https://user-images.githubusercontent.com/8281875/134416816-0a812776-b6d1-47a9-9805-866cbcf58234.png)

30 seconds of code was featured in an article published on [JavaScript in Plain English](https://javascript.plainenglish.io/11-awesome-github-repository-for-frontend-developers-c014fc8f294d).

**Comment by Chalarangelo** [comment]
#### Google Search Console Page Experience (Sep. 27th, 2021)

![image](https://user-images.githubusercontent.com/8281875/134942477-934b0b4f-9121-4027-af11-53b3addc295f.png)

**Comment by Chalarangelo** [comment]
![image](https://user-images.githubusercontent.com/8281875/134942727-be9bff72-92ea-43e7-ae12-c5e879e72949.png)

30 seconds of code was shared on Reddit [/r/webdev](https://www.reddit.com/r/webdev/comments/puj5ey/css_snippets_30_seconds_of_code/?utm_source=dlvr.it&utm_medium=twitter).

**Comment by Chalarangelo** [comment]
### September Retro

September was a great month in terms of traffic, content creation and product updates. While most of the product work was behind the scenes, we managed to deliver some major infrastructure upgrades and added some SEO and performance optimizations that will come in handy in future sprints. Content flow was consistent with our usual calendar and everything ran on schedule.

Overall, we're back up to speed, working on the content and product angles at the same time. We've developed a good tempo, we have a properly groomed backlog and we are making progress towards our strategic goals. Let's keep it up!

**Comment by Chalarangelo** [comment]
### September

**Milestone:** [2021/10](https://github.com/30-seconds/30-seconds-web/milestone/30)
**Categories:** Content, Branding
**Focus:** Content review and update, branding guide
**Description:** We are going to review and update the top 20-25% of our content (by traffic volume). This will help us create content that is more accessible (i.e. easier to understand) and more consistent. This is part of a brand image update, which will also be formalized and solidified as we work on our branding guide.
**Notes:** The branding guidelines/design system are the core of our focus at this time. However, we might not deliver 100% of it, but a lot of progress must be made, along with actionables coming from the work we put in.

**Comment by Chalarangelo** [comment]
![image](https://user-images.githubusercontent.com/8281875/135572862-85d14e63-bfc9-4bc2-aa18-c54a7c69367a.png)

30 seconds of code gained 4k stars in the past month.

**Comment by Chalarangelo** [comment]
#### Ubersuggest data (Oct. 1st, 2021)

![image](https://user-images.githubusercontent.com/8281875/135573280-35fdbf69-7d48-463a-8804-fdbe8b94da86.png)

**Comment by Chalarangelo** [comment]
### Tooling update

We have recently began the migration to a private board outside of GitHub. This means that by the end of 2021, the current model of operations will change _slightly_. Issues, infrastructure planning and internal organization will not be public anymore. We will also probably stop sharing our sprint goals and specifics for the roadmap as we go along. Instead, we will organize our work and goals off-site and deliver content and code on GitHub. This should not affect anyone in particular and it will definitely not disrupt any operations. There might, however, be some reorganization of issues that are part of a long-term backlog to clean up the repositories as well.

**Comment by Chalarangelo** [comment]
As mentioned in the previous comment, the roadmap is a relic of the past. Moving forward, planning and goals will not be public any more. The plan for 2022 is to completely embrace the new methodology, using external tools to plan ahead and track milestones. Content flow will remain consistent for the most part (~1 article/week, snippets as we see fit) and product updates will resume shortly, starting with some internal changes that need to be made to enable further updates.
